install/minikube-installation-guide.md

```diff
@@ -0,0 +1,236 @@
+| --container-runtime=cri-o | Using CRI-O for Kata |
+| --bootstrapper=kubeadm | As recommended for [minikube CRI-o](https://kubernetes.io/docs/setup/minikube/#cri-o) |
+
+> **Note:** This example deploys `Kata Containers` using `CRI-O`. The same procedure should also work
+> for a `containerd` based `k8s` installation, but is not covered in this document.
+
+> **Note:** Prior to `Minikube`/`Kubernetes` v1.14, the beta `RuntimeClass` feature also needed enabling with
+> the following. Recent `Minikube`/`Kubernetes` installations should no longer require this setting.
+>
+> | what | why |
+> | ---- | --- |
+> | --feature-gates=RuntimeClass=true | Kata needs to use the RuntimeClass k8s feature |
+
+The full command is therefore:
+
+```sh
+$ minikube start --vm-driver kvm2 --memory 6144 --network-plugin=cni --enable-default-cni --container-runtime=cri-o --bootstrapper=kubeadm 
+```
+
+## Checking for nested virtualization
+
+For `Kata Containers` to work under a `Minikube` VM, your host system must support
+nested virtualization. If you are using a Linux system utilising Intel `VT-x` 
+and the `kvm_intel` driver, you can perform the check by:
+
+```sh
+$ cat /sys/module/kvm_intel/parameters/nested
+Y
+```
+
+If your system does not report `Y` from the `nested` parameter, then details on how
+to enable nested virtualisation can be found on the
+[KVM Nested Guests page](https://www.linux-kvm.org/page/Nested_Guests)
+
+## Check `Minikube` is running
+
+Before you install `Kata Containers`, perform a quick check that your `Minikube` is
+operating. On your guest:
+
+
+```sh
+$ kubectl get nodes
+NAME       STATUS   ROLES    AGE   VERSION
+minikube   Ready    master   78m   v1.13.4
+```
+
+Check you have virtualization enabled inside your `Minikube`:
+
+```sh
+minikube ssh
+
+            _         _ ( )           ( )           
+  ___ ___  (_)  ___  (_)| |/')  _   _ | |_      __  
+/' _ ` _ `\| |/' _ `\| || , <  ( ) ( )| '_`\  /'__`\
+| ( ) ( ) || || ( ) || || |\`\ | (_) || |_) )(  ___/
+(_) (_) (_)(_)(_) (_)(_)(_) (_)`\___/'(_,__/'`\____)
+
+# No color option, as minikube is running busybox
+$ egrep 'vmx|svm' /proc/cpuinfo
+# if you get a long line of output here, you have it enabled!
+
+$ exit
+```
+
+## Installing `Kata Containers`
+
+Now you can install the `Kata` runtime components. You will need a local copy of some `Kata`
+components to help with this, and then use the host `kubectl` (that minikube has already
+configured for you) to deploy them:
+
+```sh
+$ git clone https://github.com/kata-containers/packaging.git
+$ cd packaging/kata-deploy
+$ kubectl apply -f kata-rbac.yaml
+$ kubectl apply -f kata-deploy.yaml
+```
+
+This installs the `Kata` components into `/opt/kata` inside the `Minikube` node. It can take
+a few minutes for the operation to complete. Check the installation has worked using:
+
+```sh
+$ minikube ssh
+$ cd /opt/kata
+# If the following directory is not yet created and populated, wait a short while and
```

Comment:
@grahamwhaley You can instead suggest here to run `watch kubectl get pods -n kubesystem` ans wait for the kata-deploy-xxxxx pod status to change to Running,

Reply:
Yeah, I've been hankering after a better way to wait for `kata-deploy` to finish. I was considering something like:

```
$ kubectl wait --for=condition=Ready -n kube-system DaemonSet kata-deploy
```

but.... that seems to just hang and wait... Anybody know the correct condition to wait for to check that kata-deploy has actually finished its work?

Reply:
@grahamwhaley this is how kata-deploy testing currently works
https://github.com/egernst/kata-deploy/blob/master/kata-deploy/action/test-kata.sh#L81-L82